backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/KonanDriver.kt

```diff
@@ -29,6 +29,7 @@ fun runTopLevelPhases(konanConfig: KonanConfig, environment: KotlinCoreEnvironme
         toplevelPhase.cast<CompilerPhase<Context, Unit, Unit>>().invokeToplevel(context.phaseConfig, context, Unit)
     } finally {
         context.disposeLlvm()
+        context.freeNativeMem()
```

Comment:
Please consider making two `finally` blocks here, to make sure that native memory will be freed even if `disposeLlvm` is failed.